generic/sosreport.py

```diff
@@ -0,0 +1,145 @@
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class sosreport_test(Test):
+
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        self.log.info("executing ============== %s =================" % cmd)
+        if process.system(cmd, ignore_status=True, sudo=True):
+            self.log.info("%s command failed" % cmd)
+            self.is_fail += 1
+        return
+
+    def setUp(self):
+        if "ppc" not in os.uname()[4]:
+            self.skip("supported only on Power platform")
+        dist = distro.detect()
+        sm = SoftwareManager()
+        if dist.name == 'Ubuntu':
+            if not sm.check_installed("sosreport") and not sm.install("sosreport"):
+                self.skip("Fail to install sosreport required for this test.")
+        if dist.name == 'redhat':
+            if not sm.check_installed("sos") and not sm.install("sos"):
+                self.skip("Fail to install sos package required for this test.")
+        if distro.detect().name == 'SuSE':
+            self.skip('sosreport is not supported on SLES')
+
+    def test(self):
+        self.log.info(
+            "===============Executing sosreport tool test===============")
+        self.run_cmd("sosreport -h")
+        self.run_cmd("sosreport -l")
+        self.run_cmd("sosreport --batch --case-id=test123 -n memory,samba")
+        if process.system_output("sosreport --batch --case-id=test123 -n libraries | "
+                                 "grep libraries", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--skip-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -e ntp,numa,snmp")
+        if 'sendmail' not in process.system_output("sosreport --batch --case-id=test123 -e sendmail | "
+                                                   "grep sendmail", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--enable-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -o pci,powerpc,procenv,process,processor,kdump")
+        if process.system_output("sosreport --batch --case-id=test123 -o cups | "
+                                 "grep kernel", shell=True, ignore_status=True):
+            self.is_fail += 1
+            self.log.info("--only-plugins option failed")
+        self.run_cmd("sosreport --batch --case-id=test123 -o dlm -k dlm.lockdump")
+        self.run_cmd("sosreport --batch --case-id=test123 -k filesys.dumpe2fs=on")
+        self.run_cmd("sosreport --batch --case-id=test123 --log-size=2000")
+        self.run_cmd("sosreport --batch --case-id=test123 -a")
+        self.run_cmd("sosreport --batch --case-id=test123 --all-logs")
+        self.run_cmd("sosreport --batch --case-id=test123 --build")
```

Comment:
for opt in ["-o dlm -k dlm.lockdump", "-k filesys.dumpe2fs=on", ...]:
    self.run_cmd("sosreport --batch --case-id=test123 %s" % opt)